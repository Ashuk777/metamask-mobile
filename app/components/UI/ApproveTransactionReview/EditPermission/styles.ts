import { StyleSheet } from 'react-native';

import Device from '../../../../util/device';
import { fontStyles } from '../../../../styles/common';

const createStyles = (colors: any) =>
  StyleSheet.create({
    wrapper: {
      paddingHorizontal: 24,
      paddingTop: 24,
      paddingBottom: Device.isIphoneX() ? 48 : 24,
      backgroundColor: colors.background.default,
      borderTopRightRadius: 20,
      borderTopLeftRadius: 20,
    },
    sectionExplanationText: {
      ...fontStyles.normal,
      fontSize: 12,
      color: colors.text.alternative,
      marginVertical: 6,
    },
    option: {
      flexDirection: 'row',
      marginVertical: 8,
    },
    errorMessageWrapper: {
      marginVertical: 6,
    },
    optionText: {
      ...fontStyles.normal,
      fontSize: 14,
      lineHeight: 20,
      color: colors.text.default,
    },
    touchableOption: {
      flexDirection: 'row',
    },
    selectedCircle: {
      width: 8,
      height: 8,
      borderRadius: 8 / 2,
      margin: 3,
      backgroundColor: colors.primary.default,
    },
    outSelectedCircle: {
      width: 18,
      height: 18,
      borderRadius: 18 / 2,
      borderWidth: 2,
      borderColor: colors.primary.default,
    },
    circle: {
      width: 18,
      height: 18,
      borderRadius: 18 / 2,
      backgroundColor: colors.background.default,
      opacity: 1,
      borderWidth: 2,
      borderColor: colors.border.default,
    },
    input: {
      padding: 12,
      borderColor: colors.border.default,
      borderRadius: 10,
      borderWidth: 2,
      color: colors.text.default,
    },
    spendLimitContent: {
      marginLeft: 8,
      flex: 1,
    },
    spendLimitTitle: {
      ...fontStyles.bold,
      color: colors.text.default,
      fontSize: 14,
      lineHeight: 20,
      marginBottom: 8,
    },
    spendLimitSubtitle: {
      ...fontStyles.normal,
      fontSize: 12,
      lineHeight: 18,
      color: colors.text.alternative,
    },
    textBlue: {
      color: colors.primary.default,
    },
    textBlack: {
      color: colors.text.default,
    },
  });

export default createStyles;
