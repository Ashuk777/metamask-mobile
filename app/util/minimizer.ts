import { Platform } from 'react-native';
import Minimizer from 'react-native-minimizer';
import { MetamaskMinimizer } from '../core/NativeModules';

export default Platform.select({
  ios: Minimizer,
  android: MetamaskMinimizer,
});
