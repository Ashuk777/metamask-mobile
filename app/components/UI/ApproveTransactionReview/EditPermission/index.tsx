import React, { useCallback, useEffect, useState } from 'react';
import { View, Text, TouchableOpacity, TextInput } from 'react-native';
import { fontStyles } from '../../../../styles/common';
import {
  decodeApproveData,
  generateTxWithNewTokenAllowance,
  minimumTokenAllowance,
} from '../../../../util/transactions';
import StyledButton from '../../StyledButton';
import { strings } from '../../../../../locales/i18n';
import {
  fromTokenMinimalUnit,
  hexToBN,
  isNumber,
} from '../../../../util/number';
import Logger from '../../../../util/Logger';
import AnalyticsV2 from '../../../../util/analyticsV2';
import { MetaMetricsEvents } from '../../../../core/Analytics';
import ConnectHeader from '../../ConnectHeader';
import ErrorMessage from '../../../Views/SendFlow/ErrorMessage';
import { useTheme } from '../../../../util/theme';
import formatNumber from '../../../../util/formatNumber';
import createStyles from './styles';

interface IEditPermissionProps {
  host: string;
  token: { tokenSymbol: string; tokenDecimals: number };
  originalApproveAmount: string;
  spendLimitCustomValue: string;
  onSpendLimitCustomValueChange: (approvalCustomValue: string) => void;
  spendLimitUnlimitedSelected: boolean;
  onSpendLimitUnlimitedSelectedChange: (value: boolean) => void;
  transaction: Record<string, unknown> & { data: string };
  setTransactionObject: (
    transaction: Record<string, unknown> & { data: string },
  ) => void;
  toggleEditPermission: () => void;
  onCustomSpendAmountChange: (amount: string) => void;
  spenderAddress: string;
  getAnalyticsParams: () => Record<string, unknown>;
}

function EditPermission({
  host,
  token,
  originalApproveAmount,
  spendLimitCustomValue,
  onSpendLimitCustomValueChange,
  spendLimitUnlimitedSelected,
  onSpendLimitUnlimitedSelectedChange,
  transaction,
  setTransactionObject,
  toggleEditPermission,
  onCustomSpendAmountChange,
  spenderAddress,
  getAnalyticsParams,
}: IEditPermissionProps) {
  const [initialState] = useState({
    spendLimitUnlimitedSelected,
    spendLimitCustomValue,
  });
  const [disableBtn, setDisableBtn] = useState(false);
  const [displayErrorMsg, setDisplayErrorMsg] = useState(false);
  const { colors, themeAppearance } = useTheme();
  const { tokenSymbol, tokenDecimals } = token;
  const minimumSpendLimit = minimumTokenAllowance(tokenDecimals);
  const styles = createStyles(colors);

  useEffect(() => {
    setDisplayErrorMsg(
      Number(minimumSpendLimit) > Number(spendLimitCustomValue),
    );
  }, [minimumSpendLimit, spendLimitCustomValue, spendLimitUnlimitedSelected]);

  useEffect(() => {
    setDisableBtn(!isNumber(spendLimitCustomValue) || displayErrorMsg);
  }, [spendLimitCustomValue, displayErrorMsg]);

  const onPressSpendLimitUnlimitedSelected = useCallback(() => {
    onSpendLimitUnlimitedSelectedChange(true);
    onSpendLimitCustomValueChange(minimumSpendLimit);
  }, [
    onSpendLimitUnlimitedSelectedChange,
    onSpendLimitCustomValueChange,
    minimumSpendLimit,
  ]);

  const onBackPress = useCallback(() => {
    if (initialState.spendLimitUnlimitedSelected) {
      onSpendLimitUnlimitedSelectedChange(true);
    } else {
      onSpendLimitUnlimitedSelectedChange(false);
    }
    onSpendLimitCustomValueChange(initialState.spendLimitCustomValue);
    toggleEditPermission();
  }, [
    initialState,
    onSpendLimitUnlimitedSelectedChange,
    onSpendLimitCustomValueChange,
    toggleEditPermission,
  ]);

  const onSetApprovalAmount = useCallback(() => {
    if (!spendLimitUnlimitedSelected && !spendLimitCustomValue) {
      onPressSpendLimitUnlimitedSelected();
    } else {
      try {
        const newApprovalTransaction = generateTxWithNewTokenAllowance(
          spendLimitUnlimitedSelected
            ? originalApproveAmount
            : spendLimitCustomValue,
          tokenDecimals,
          spenderAddress,
          transaction,
        );

        const { encodedAmount } = decodeApproveData(
          newApprovalTransaction.data,
        );

        const approveAmount = fromTokenMinimalUnit(
          hexToBN(encodedAmount),
          tokenDecimals.toString(),
        );

        onCustomSpendAmountChange(approveAmount);
        setTransactionObject({
          ...newApprovalTransaction,
          transaction: {
            ...(newApprovalTransaction as any).transaction,
            data: newApprovalTransaction.data,
          },
        });
      } catch (err) {
        Logger.log('Failed to setTransactionObject', err);
      }
      toggleEditPermission();
      AnalyticsV2.trackEvent(
        MetaMetricsEvents.APPROVAL_PERMISSION_UPDATED,
        getAnalyticsParams(),
      );
    }
  }, [
    getAnalyticsParams,
    onCustomSpendAmountChange,
    onPressSpendLimitUnlimitedSelected,
    originalApproveAmount,
    setTransactionObject,
    spenderAddress,
    spendLimitCustomValue,
    spendLimitUnlimitedSelected,
    toggleEditPermission,
    tokenDecimals,
    transaction,
  ]);

  return (
    <View style={styles.wrapper}>
      <ConnectHeader
        action={onBackPress}
        title={strings('spend_limit_edition.title')}
      />
      <View>
        <Text style={styles.spendLimitTitle}>
          {strings('spend_limit_edition.spend_limit')}
        </Text>
        <Text style={styles.spendLimitSubtitle}>
          {strings('spend_limit_edition.allow')}
          <Text style={fontStyles.bold}>{` ${host} `}</Text>
          {strings('spend_limit_edition.allow_explanation')}
        </Text>

        <View style={styles.option}>
          <TouchableOpacity
            onPress={onPressSpendLimitUnlimitedSelected}
            style={styles.touchableOption}
          >
            {spendLimitUnlimitedSelected ? (
              <View style={styles.outSelectedCircle}>
                <View style={styles.selectedCircle} />
              </View>
            ) : (
              <View style={styles.circle} />
            )}
          </TouchableOpacity>
          <View style={styles.spendLimitContent}>
            <Text
              style={[
                styles.optionText,
                spendLimitUnlimitedSelected
                  ? styles.textBlue
                  : styles.textBlack,
              ]}
            >
              {strings('spend_limit_edition.proposed')}
            </Text>
            <Text style={styles.sectionExplanationText}>
              {strings('spend_limit_edition.requested_by')}
              <Text style={fontStyles.bold}>{` ${host}`}</Text>
            </Text>
            <Text
              style={[styles.optionText, styles.textBlack]}
            >{`${formatNumber(originalApproveAmount)} ${tokenSymbol}`}</Text>
          </View>
        </View>

        <View style={styles.option}>
          <TouchableOpacity
            onPress={onPressSpendLimitUnlimitedSelected}
            style={styles.touchableOption}
          >
            {spendLimitUnlimitedSelected ? (
              <View style={styles.circle} />
            ) : (
              <View style={styles.outSelectedCircle}>
                <View style={styles.selectedCircle} />
              </View>
            )}
          </TouchableOpacity>
          <View style={styles.spendLimitContent}>
            <Text
              style={[
                styles.optionText,
                !spendLimitUnlimitedSelected
                  ? styles.textBlue
                  : styles.textBlack,
              ]}
            >
              {strings('spend_limit_edition.custom_spend_limit')}
            </Text>
            <Text style={styles.sectionExplanationText}>
              {strings('spend_limit_edition.max_spend_limit')}
            </Text>
            <TextInput
              autoCapitalize="none"
              keyboardType="numeric"
              autoCorrect={false}
              onChangeText={onSpendLimitCustomValueChange}
              placeholder={`100 ${tokenSymbol}`}
              placeholderTextColor={colors.text.muted}
              spellCheck={false}
              style={styles.input}
              value={spendLimitCustomValue}
              numberOfLines={1}
              onFocus={() => onSpendLimitUnlimitedSelectedChange(false)}
              returnKeyType={'done'}
              keyboardAppearance={themeAppearance}
            />
            {displayErrorMsg && (
              <View style={styles.errorMessageWrapper}>
                <ErrorMessage
                  errorMessage={strings(
                    'spend_limit_edition.must_be_at_least',
                    {
                      allowance: minimumSpendLimit,
                    },
                  )}
                />
              </View>
            )}
          </View>
        </View>
      </View>
      <StyledButton
        disabled={disableBtn}
        type="confirm"
        onPress={onSetApprovalAmount}
      >
        {strings('transaction.set_gas')}
      </StyledButton>
    </View>
  );
}

export default EditPermission;
