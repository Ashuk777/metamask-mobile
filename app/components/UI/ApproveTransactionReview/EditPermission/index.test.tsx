import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import { shallow } from 'enzyme';

import { ThemeContext, mockTheme } from '../../../../util/theme';
import EditPermission from './';

const props = {
  host: 'www.google.com',
  token: { tokenSymbol: 'TDC', tokenDecimals: 6 },
  originalApproveAmount: '10',
  spendLimitCustomValue: '.0001',
  onSpendLimitCustomValueChange: () => undefined,
  spendLimitUnlimitedSelected: false,
  onSpendLimitUnlimitedSelectedChange: () => undefined,
  transaction: { data: '0x0' },
  setTransactionObject: () => undefined,
  toggleEditPermission: () => undefined,
  onCustomSpendAmountChange: () => undefined,
  spenderAddress: '0x0',
  getAnalyticsParams: () => ({}),
};

describe('EditPermission', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<EditPermission {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot', () => {
    const container = render(
      <ThemeContext.Provider value={mockTheme}>
        <EditPermission {...props} />
      </ThemeContext.Provider>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should call onCustomSpendAmountChange when "Set" button in footer is clicked', () => {
    const onCustomSpendAmountChange = jest.fn();
    const { getByRole } = render(
      <ThemeContext.Provider value={mockTheme}>
        <EditPermission
          {...props}
          onCustomSpendAmountChange={onCustomSpendAmountChange}
        />
      </ThemeContext.Provider>,
    );
    fireEvent.press(getByRole('button', { name: 'Set' }));
    expect(onCustomSpendAmountChange).toBeCalledTimes(1);
  });
});
