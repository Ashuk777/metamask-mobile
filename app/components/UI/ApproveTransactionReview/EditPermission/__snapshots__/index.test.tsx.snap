// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditPermission should match snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderTopLeftRadius": 20,
      "borderTopRightRadius": 20,
      "paddingBottom": 48,
      "paddingHorizontal": 24,
      "paddingTop": 24,
    }
  }
>
  <View
    style={
      Object {
        "paddingBottom": 20,
        "position": "relative",
        "width": "100%",
      }
    }
  >
    <TouchableOpacity
      onPress={[Function]}
      style={
        Object {
          "paddingRight": 10,
          "paddingVertical": 10,
          "position": "absolute",
          "zIndex": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#24272A",
              "fontSize": 24,
            },
            undefined,
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        Ôèè
      </Text>
    </TouchableOpacity>
    <Text
      style={
        Object {
          "color": "#24272A",
          "fontFamily": "Euclid Circular B",
          "fontSize": 16,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 24,
          "paddingVertical": 12,
          "textAlign": "center",
        }
      }
    >
      Edit permission
    </Text>
  </View>
  <View>
    <Text
      style={
        Object {
          "color": "#24272A",
          "fontFamily": "EuclidCircularB-Bold",
          "fontSize": 14,
          "fontWeight": "600",
          "lineHeight": 20,
          "marginBottom": 8,
        }
      }
    >
      Spend limit permission
    </Text>
    <Text
      style={
        Object {
          "color": "#535A61",
          "fontFamily": "EuclidCircularB-Regular",
          "fontSize": 12,
          "fontWeight": "400",
          "lineHeight": 18,
        }
      }
    >
      Allow
      <Text
        style={
          Object {
            "fontFamily": "EuclidCircularB-Bold",
            "fontWeight": "600",
          }
        }
      >
         www.google.com 
      </Text>
      to withdraw and spend up to the following amount:
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginVertical": 8,
        }
      }
    >
      <TouchableOpacity
        onPress={[Function]}
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#BBC0C5",
              "borderRadius": 9,
              "borderWidth": 2,
              "height": 18,
              "opacity": 1,
              "width": 18,
            }
          }
        />
      </TouchableOpacity>
      <View
        style={
          Object {
            "flex": 1,
            "marginLeft": 8,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#24272A",
                "fontFamily": "EuclidCircularB-Regular",
                "fontSize": 14,
                "fontWeight": "400",
                "lineHeight": 20,
              },
              Object {
                "color": "#24272A",
              },
            ]
          }
        >
          Proposed approval limit
        </Text>
        <Text
          style={
            Object {
              "color": "#535A61",
              "fontFamily": "EuclidCircularB-Regular",
              "fontSize": 12,
              "fontWeight": "400",
              "marginVertical": 6,
            }
          }
        >
          Spend limit requested by
          <Text
            style={
              Object {
                "fontFamily": "EuclidCircularB-Bold",
                "fontWeight": "600",
              }
            }
          >
             www.google.com
          </Text>
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#24272A",
                "fontFamily": "EuclidCircularB-Regular",
                "fontSize": 14,
                "fontWeight": "400",
                "lineHeight": 20,
              },
              Object {
                "color": "#24272A",
              },
            ]
          }
        >
          10 TDC
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginVertical": 8,
        }
      }
    >
      <TouchableOpacity
        onPress={[Function]}
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "borderColor": "#0376C9",
              "borderRadius": 9,
              "borderWidth": 2,
              "height": 18,
              "width": 18,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#0376C9",
                "borderRadius": 4,
                "height": 8,
                "margin": 3,
                "width": 8,
              }
            }
          />
        </View>
      </TouchableOpacity>
      <View
        style={
          Object {
            "flex": 1,
            "marginLeft": 8,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#24272A",
                "fontFamily": "EuclidCircularB-Regular",
                "fontSize": 14,
                "fontWeight": "400",
                "lineHeight": 20,
              },
              Object {
                "color": "#0376C9",
              },
            ]
          }
        >
          Custom spend limit
        </Text>
        <Text
          style={
            Object {
              "color": "#535A61",
              "fontFamily": "EuclidCircularB-Regular",
              "fontSize": 12,
              "fontWeight": "400",
              "marginVertical": 6,
            }
          }
        >
          Enter a max spend limit
        </Text>
        <TextInput
          autoCapitalize="none"
          autoCorrect={false}
          keyboardAppearance="light"
          keyboardType="numeric"
          numberOfLines={1}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder="100 TDC"
          placeholderTextColor="#BBC0C5"
          returnKeyType="done"
          spellCheck={false}
          style={
            Object {
              "borderColor": "#BBC0C5",
              "borderRadius": 10,
              "borderWidth": 2,
              "color": "#24272A",
              "padding": 12,
            }
          }
          value=".0001"
        />
      </View>
    </View>
  </View>
  <TouchableOpacity
    accessibilityRole="button"
    activeOpacity={1}
    disabled={true}
    style={
      Array [
        Array [
          Object {
            "borderRadius": 100,
            "justifyContent": "center",
            "padding": 15,
          },
          Object {
            "backgroundColor": "#0376C9",
            "minHeight": 50,
          },
          undefined,
        ],
        Object {
          "opacity": 0.6,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#007aff",
            "fontSize": 17,
            "fontWeight": "500",
            "textAlign": "center",
          },
          Object {
            "color": "#dcdcdc",
          },
          Array [
            Object {
              "fontFamily": "EuclidCircularB-Bold",
              "fontSize": 14,
              "fontWeight": "600",
              "textAlign": "center",
            },
            Object {
              "color": "#FCFCFC",
            },
            undefined,
          ],
          Object {
            "opacity": 0.6,
          },
        ]
      }
    >
      Set
    </Text>
  </TouchableOpacity>
</View>
`;

exports[`EditPermission should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderTopLeftRadius": 20,
      "borderTopRightRadius": 20,
      "paddingBottom": 48,
      "paddingHorizontal": 24,
      "paddingTop": 24,
    }
  }
>
  <ConnectHeader
    action={[Function]}
    title="Edit permission"
  />
  <View>
    <Text
      style={
        Object {
          "color": "#24272A",
          "fontFamily": "EuclidCircularB-Bold",
          "fontSize": 14,
          "fontWeight": "600",
          "lineHeight": 20,
          "marginBottom": 8,
        }
      }
    >
      Spend limit permission
    </Text>
    <Text
      style={
        Object {
          "color": "#535A61",
          "fontFamily": "EuclidCircularB-Regular",
          "fontSize": 12,
          "fontWeight": "400",
          "lineHeight": 18,
        }
      }
    >
      Allow
      <Text
        style={
          Object {
            "fontFamily": "EuclidCircularB-Bold",
            "fontWeight": "600",
          }
        }
      >
         www.google.com 
      </Text>
      to withdraw and spend up to the following amount:
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginVertical": 8,
        }
      }
    >
      <TouchableOpacity
        onPress={[Function]}
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#BBC0C5",
              "borderRadius": 9,
              "borderWidth": 2,
              "height": 18,
              "opacity": 1,
              "width": 18,
            }
          }
        />
      </TouchableOpacity>
      <View
        style={
          Object {
            "flex": 1,
            "marginLeft": 8,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#24272A",
                "fontFamily": "EuclidCircularB-Regular",
                "fontSize": 14,
                "fontWeight": "400",
                "lineHeight": 20,
              },
              Object {
                "color": "#24272A",
              },
            ]
          }
        >
          Proposed approval limit
        </Text>
        <Text
          style={
            Object {
              "color": "#535A61",
              "fontFamily": "EuclidCircularB-Regular",
              "fontSize": 12,
              "fontWeight": "400",
              "marginVertical": 6,
            }
          }
        >
          Spend limit requested by
          <Text
            style={
              Object {
                "fontFamily": "EuclidCircularB-Bold",
                "fontWeight": "600",
              }
            }
          >
             www.google.com
          </Text>
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#24272A",
                "fontFamily": "EuclidCircularB-Regular",
                "fontSize": 14,
                "fontWeight": "400",
                "lineHeight": 20,
              },
              Object {
                "color": "#24272A",
              },
            ]
          }
        >
          10 TDC
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginVertical": 8,
        }
      }
    >
      <TouchableOpacity
        onPress={[Function]}
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "borderColor": "#0376C9",
              "borderRadius": 9,
              "borderWidth": 2,
              "height": 18,
              "width": 18,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#0376C9",
                "borderRadius": 4,
                "height": 8,
                "margin": 3,
                "width": 8,
              }
            }
          />
        </View>
      </TouchableOpacity>
      <View
        style={
          Object {
            "flex": 1,
            "marginLeft": 8,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#24272A",
                "fontFamily": "EuclidCircularB-Regular",
                "fontSize": 14,
                "fontWeight": "400",
                "lineHeight": 20,
              },
              Object {
                "color": "#0376C9",
              },
            ]
          }
        >
          Custom spend limit
        </Text>
        <Text
          style={
            Object {
              "color": "#535A61",
              "fontFamily": "EuclidCircularB-Regular",
              "fontSize": 12,
              "fontWeight": "400",
              "marginVertical": 6,
            }
          }
        >
          Enter a max spend limit
        </Text>
        <TextInput
          autoCapitalize="none"
          autoCorrect={false}
          keyboardAppearance="light"
          keyboardType="numeric"
          numberOfLines={1}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder="100 TDC"
          placeholderTextColor="#BBC0C5"
          returnKeyType="done"
          spellCheck={false}
          style={
            Object {
              "borderColor": "#BBC0C5",
              "borderRadius": 10,
              "borderWidth": 2,
              "color": "#24272A",
              "padding": 12,
            }
          }
          value=".0001"
        />
      </View>
    </View>
  </View>
  <StyledButton
    disabled={false}
    disabledContainerStyle={
      Object {
        "opacity": 0.6,
      }
    }
    onPress={[Function]}
    styleDisabled={
      Object {
        "opacity": 0.6,
      }
    }
    type="confirm"
  >
    Set
  </StyledButton>
</View>
`;
